### ISSUE TITLE
revdeps broken by DT[, j=<expr>:<expr] being with=TRUE

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#6486]
A number of revdeps were broken by db3cbb47212590ef1d1b0e2ce797178fc2e1baaf (#4460):

```
MetaIntegrator
SOMnmR
VoxR
categoryEncodings
flextable
hesim
iml
immunarch
kalmanfilter
polypharmacy
sentometrics
sjtable2df
```

I haven't looked into the root cause(s) yet.

**Comment by MichaelChirico** [comment]
OK, here's an MRE (not sure if it covers everything):

```r
DT=data.table(a=1,b=2)
DT[,1:ncol(DT)]

# ** on CRAN
#        a     b
#    <num> <num>
# 1:     1     2

# ** on master
DT[, 1:ncol(DT)]
# [1] 1 2
```

At a glance, we need to be more selective & make sure `all(all.vars(jsub) %in% names(x))` before switching to `with=TRUE`.

**Comment by MichaelChirico** [comment]
#6487 solved everything, nice 😎

**Comment by jangorecki** [comment]
@MichaelChirico If this fix is also going to be in 1.16.2 (I haven't see it merged to patch branch, but feels like it fits there, no?) then we should assign it to that milestone instead.

**Comment by MichaelChirico** [comment]
no, original breakage #6487 was only in dev
